### PR TITLE
Progress update

### PR DESCRIPTION
**Pull request by ericnewcomer**

### CODE REVIEW

File: temba/msgs/views.py [hunk 1]
```diff
@@ -717,6 +719,48 @@ def form_valid(self, form):
...
+                        "created_on": obj.created_on.isoformat(),
+                        "modified_on": obj.modified_on.isoformat(),
+                        "progress": {"total": obj.contact_count, "current": obj.get_message_count()},
+                    }
+                )
+            return JsonResponse({"results": results})
+
+    class Interrupt(ModalMixin, OrgObjPermsMixin, SmartUpdateView):
+        default_template = "smartmin/delete_confirm.html"
+        permission = "msgs.broadcast_create"
```
Comment (rowanseymour) [inline temba/msgs/views.py:754]: `"msgs.broadcast_update"`

File: temba/msgs/views.py [hunk 2]
```diff
@@ -717,6 +719,48 @@ def form_valid(self, form):
 
             return self.render_modal_response(form)
 
+    class Status(OrgPermsMixin, SmartListView):
+        permission = "msgs.broadcast_create"
+
+        def derive_queryset(self, **kwargs):
+            qs = super().derive_queryset(**kwargs)
```
Comment (rowanseymour) [inline temba/msgs/views.py:726]: this isn't filtering by org no?

File: temba/msgs/views.py [hunk 3]
```diff
@@ -717,6 +719,48 @@ def form_valid(self, form):
 
             return self.render_modal_response(form)
 
+    class Status(OrgPermsMixin, SmartListView):
+        permission = "msgs.broadcast_create"
```
Comment (rowanseymour) [inline temba/msgs/views.py:723]: `"msgs.broadcast_read"`

File: temba/settings_common.py [hunk 4]
```diff
@@ -359,10 +359,11 @@
     "contacts.contactgroup": ("menu",),
     "contacts.contactimport": ("preview",),
     "flows.flow": ("assets", "copy", "editor", "export", "menu", "results", "start"),
+    "flows.flowstart": ("interrupt", "status"),
     "flows.flowsession": ("json",),
     "globals.global": ("unused",),
     "locations.adminboundary": ("alias", "boundaries", "geometry"),
-    "msgs.broadcast": ("scheduled", "scheduled_read", "scheduled_delete"),
+    "msgs.broadcast": ("scheduled", "scheduled_read", "scheduled_delete", "interrupt", "status"),
```
Comment (rowanseymour) [inline temba/settings_common.py:366]: not needed

File: temba/flows/views.py [hunk 5]
```diff
@@ -1875,3 +1875,52 @@ def get_context_data(self, *args, **kwargs):
...
+                            "uuid": obj.flow.uuid,
+                        },
+                        "progress": {"total": obj.contact_count, "current": obj.run_count},
+                    }
+                )
+            return JsonResponse({"results": results})
+
+    class Interrupt(ModalMixin, OrgObjPermsMixin, SmartUpdateView):
+        default_template = "smartmin/delete_confirm.html"
+        permission = "flows.flow_start"
```
Comment (rowanseymour) [inline temba/flows/views.py:1918]: `"flows.flowstart_update"`

File: temba/flows/views.py [hunk 6]
```diff
@@ -1875,3 +1875,52 @@ def get_context_data(self, *args, **kwargs):
             FlowStartCount.bulk_annotate(context["object_list"])
 
             return context
+
+    class Status(OrgPermsMixin, SmartListView):
+        permission = "flows.flow_start"
```
Comment (rowanseymour) [inline temba/flows/views.py:1880]: `"flows.flowstart_read"`

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/orgs/views.py [hunk 7]
```diff
@@ -138,9 +138,7 @@ def has_permission(self, request, *args, **kwargs):
         self.args = args
         self.request = request
 
-        org = self.derive_org()
-
-        if self.get_user().is_staff and org:
```
Comment (ericnewcomer) [inline temba/orgs/views.py:143]: @rowanseymour so this touches permissions all over the place, but I'm not sure why org is required here. Any idea? This is to allow staff users to use the status endpoint, etc without an active org.

File: templates/frame.html [hunk 8]
```diff
@@ -211,8 +211,6 @@
                 </temba-menu>
               </div>
               <div class="flex-grow spa-container flex flex-col initial-load">
-                <temba-outbox-monitor>
-                </temba-outbox-monitor>
```
Comment (ericnewcomer) [inline templates/frame.html:215]: no point in this thing anymore

File: temba/flows/views.py [hunk 9]
```diff
@@ -1691,12 +1691,17 @@ def derive_initial(self):
                     urn = urn.get_display(org=org, international=True)
                 recipients.append({"id": contact.uuid, "name": contact.name, "urn": urn, "type": "contact"})
 
+            exclusions = settings.DEFAULT_EXCLUSIONS.copy()
+
+            if self.flow and self.flow.flow_type == Flow.TYPE_BACKGROUND:
+                del exclusions["in_a_flow"]
+
```
Comment (ericnewcomer) [inline temba/flows/views.py:1698]: Snuck this in. Background flows while not showing the in_a_flow checkbox could have it "enabled" by default.

**Pull request merged**